src/Perpetuum/Services/EventServices/EventProcessors/NpcStateAnnouncer.cs

```diff
@@ -51,10 +52,25 @@ private int GetNpcDef(NpcStateMessage msg)
         private string GetNpcName(int def)
         {
             var nameToken = EntityDefault.Get(def).Name + "_name";
-            var name = _nameDictionary[nameToken]?.ToString();
+            var name = "";
+            if (_nameDictionary.TryGetValue(nameToken, out name) != true)
+            {
+                WriteNPCStateAnnouncerLog($"Missing localized name for {nameToken}");
+            }
             return name ?? string.Empty;
         }
 
+        private void WriteNPCStateAnnouncerLog(string message)
+        {
+            var e = new LogEvent
+            {
+                LogType = LogType.Info,
+                Tag = "NPC",
+                Message = message
+            };
+
+            Logger.Log(e);
```

Comment:
I'd be ok with a simple string format in a `Logger.Warn($"...")` but this is good too.
But do at least make it warn level.  Bad/missing config that doesn't break things should be a warning at least, in my opinion.  Then anyone crawling logs for things to fix can just look at Warn and above for issues :)

Reply:
Makes sense. Thoughts on the tagging? "NPC" might be a bit broad.

Reply:
Yes, I suppose the name of this class would do?